sdk/src/main/java/com/horizen/proposition/PublicKey25519PropositionSerializer.java

```diff
@@ -27,4 +30,13 @@ public void serialize(PublicKey25519Proposition proposition, Writer writer) {
     public PublicKey25519Proposition parse(Reader reader) {
         return new PublicKey25519Proposition(reader.getBytes(PublicKey25519Proposition.KEY_LENGTH));
     }
+
+    public PublicKey25519Proposition parseAndCheck(Reader reader) {
```

Comment:
similar here:
`parse(Reader reader)` may just pass to the method `parse(reader, false)`
`parseAndCheck` -> `parse`

Reply:
We cannot change method parse interface cause it is overloaded. Interface of this method is defined in scorex's Serializer.

Reply:
Summary after discussion:
1) parseAndCheck(Reader reader) -> `parse(Reader reader, boolean checkPubKey)`
2) parse(Reader reader) -> inside call `parse(reader, false)`
3) parseBytesAndCheck -> inside cal `return parse(bufferReader, true);`
4) the same for Schnorr and VRF **public** keys